### PR TITLE
Track basic Kubernetes resources in Pulumi from Helm

### PR DESCRIPTION
**Pull request by RossComputerGuy**
This PR introduces a fix for Tilt which allows for proper resource tracking so Tilt will give a proper "ready" state. This allows for stacks with only Helm resources to work properly. I have not yet got Tilt to test it out but I have tested it in the terminal and it does work there.

Resources which now are tracked:

* `ConfigMap`
* `Deployment`
* `Secret`
* `Service`
* `ServiceAccount`
* `StatefulSets`

Anyone is welcomed to add new types by adding the key to the `helmResourceKindMap` variable. However, it must follow how Pulumi defines it in `resourceNames` and it must be set to a valid lowercase resource name. In the future, I might want to have a more "intuitive" way which doesn't require a dictionary.

### CODE REVIEW

**Review by nicks — COMMENTED** [review]
we recently added a DCO bot, so you'll need to sign your commits. thanks! https://github.com/tilt-dev/tilt-extensions/pull/430/checks?check_run_id=7600465054

File: pulumi/pulumi-get.py [hunk 1]
```diff
@@ -7,6 +7,25 @@
 import subprocess
 import sys
 
+helmResourceKindMap = {
```
[the remark this answers is not in the document]
Comment (nicks) [reply pulumi/pulumi-get.py:10]: i think it's fine to lowercase the first part of the name, that should usually work

File: pulumi/pulumi-get.py [hunk 2]
```diff
@@ -19,15 +38,28 @@
...
       continue
 
     kind = o.get('kind', '').lower()
     name = o.get('metadata', {}).get('name', '')
-    namespace = o.get('metadata', {}).get('namespace', '')
-    if kind and name:
-      print(subprocess.check_output(['kubectl', 'get', '-n', namespace, kind, name, '-o=yaml']).decode('utf-8'))
-      print('---')
+    ns = o.get('metadata', {}).get('namespace', '')
+    dumpResource(ns, kind, name)
```
[the remark this answers is not in the document]
Comment (RossComputerGuy) [reply pulumi/pulumi-get.py:65]: That check is done in the new `dumbResource` function.

File: pulumi/pulumi-get.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RossComputerGuy) [reply pulumi/pulumi-get.py:10]: This method is much simpler to convert the resource types Pulumi defines, plus it allows for pushing the resources that are most critical. Some examples of how Pulumi has it defined:
```json
{
  "Deployment.apps/apps/v1": [
    "cerusbots/cerus-mailhog"
  ],
  "Service/v1": [
    "cerusbots/cerus-mailhog"
  ],
  "ServiceAccount/v1": [
    "cerusbots/cerus-mailhog"
  ]
}
```
```json
{
  "ConfigMap/v1": [
    "cerusbots/cerus-grafana-envvars"
  ],
  "Deployment.apps/apps/v1": [
    "cerusbots/cerus-grafana"
  ],
  "PersistentVolumeClaim/v1": [
    "cerusbots/cerus-grafana"
  ],
  "Secret/v1": [
    "cerusbots/cerus-grafana-admin"
  ],
  "Service/v1": [
    "cerusbots/cerus-grafana"
  ],
  "ServiceAccount/v1": [
    "cerusbots/cerus-grafana"
  ]
}
```
```json
{
  "ConfigMap/v1": [
    "cerus-analytics-clickhouse-config",
    "cerus-analytics-clickhouse-metrica",
    "cerus-analytics-clickhouse-users"
  ],
  "Deployment.apps/apps/v1": [
    "cerus-analytics-plausible-analytics"
  ],
  "Secret/v1": [
    "cerus-analytics-plausible-analytics",
    "cerusbots/cerus-analytics-postgresql"
  ],
  "Service/v1": [
    "cerus-analytics-clickhouse",
    "cerus-analytics-clickhouse-headless",
    "cerus-analytics-clickhouse-replica",
    "cerus-analytics-clickhouse-replica-headless",
    "cerus-analytics-plausible-analytics",
    "cerusbots/cerus-analytics-postgresql",
    "cerusbots/cerus-analytics-postgresql-hl"
  ],
  "ServiceAccount/v1": [
    "cerus-analytics-plausible-analytics"
  ],
  "StatefulSet.apps/apps/v1": [
    "cerus-analytics-clickhouse",
    "cerus-analytics-clickhouse-replica",
    "cerusbots/cerus-analytics-postgresql"
  ]
}
```
```json
{
  "Alertmanager.monitoring.coreos.com/monitoring.coreos.com/v1": [
    "cerusbots/cerus-prometheus-kube-prom-alertmanager"
  ],
  "ClusterRole.rbac.authorization.k8s.io/rbac.authorization.k8s.io/v1": [
    "cerusbots/cerus-prometheus-kube-prom-alertmanager-psp",
    "cerusbots/cerus-prometheus-kube-prom-operator",
    "cerusbots/cerus-prometheus-kube-prom-operator-psp",
    "cerusbots/cerus-prometheus-kube-prom-prometheus",
    "cerusbots/cerus-prometheus-kube-prom-prometheus-psp",
    "cerusbots/cerus-prometheus-kube-state-metrics-cerusbots",
    "cerusbots/cerus-prometheus-kube-state-metrics-cerusbots-psp",
    "cerusbots/cerus-prometheus-node-exporter-psp"
  ],
  "ClusterRoleBinding.rbac.authorization.k8s.io/rbac.authorization.k8s.io/v1": [
    "cerusbots/cerus-prometheus-kube-prom-alertmanager-psp",
    "cerusbots/cerus-prometheus-kube-prom-operator",
    "cerusbots/cerus-prometheus-kube-prom-operator-psp",
    "cerusbots/cerus-prometheus-kube-prom-prometheus",
    "cerusbots/cerus-prometheus-kube-prom-prometheus-psp",
    "cerusbots/cerus-prometheus-kube-state-metrics-cerusbots",
    "cerusbots/cerus-prometheus-kube-state-metrics-cerusbots-psp",
    "cerusbots/cerus-prometheus-node-exporter-psp"
  ],
  "ConfigMap/v1": [
    "cerusbots/cerus-prometheus-kube-prom-blackbox-exporter",
    "cerusbots/cerus-prometheus-kube-prom-operator"
  ],
  "DaemonSet.apps/apps/v1": [
    "cerusbots/cerus-prometheus-node-exporter"
  ],
  "Deployment.apps/apps/v1": [
    "cerusbots/cerus-prometheus-kube-prom-blackbox-exporter",
    "cerusbots/cerus-prometheus-kube-prom-operator",
    "cerusbots/cerus-prometheus-kube-state-metrics"
  ],
  "PodSecurityPolicy.policy/policy/v1beta1": [
    "cerusbots/cerus-prometheus-kube-prom-alertmanager",
    "cerusbots/cerus-prometheus-kube-prom-operator",
    "cerusbots/cerus-prometheus-kube-prom-prometheus",
    "cerusbots/cerus-prometheus-kube-state-metrics-cerusbots",
    "cerusbots/cerus-prometheus-node-exporter"
  ],
  "Prometheus.monitoring.coreos.com/monitoring.coreos.com/v1": [
    "cerusbots/cerus-prometheus-kube-prom-prometheus"
  ],
  "Secret/v1": [
    "cerusbots/alertmanager-cerus-prometheus-kube-prom-alertmanager"
  ],
  "Service/v1": [
    "cerusbots/cerus-prometheus-kube-prom-alertmanager",
    "cerusbots/cerus-prometheus-kube-prom-blackbox-exporter",
    "cerusbots/cerus-prometheus-kube-prom-operator",
    "cerusbots/cerus-prometheus-kube-prom-prometheus",
    "cerusbots/cerus-prometheus-kube-state-metrics",
    "cerusbots/cerus-prometheus-node-exporter",
    "kube-system/cerus-prometheus-kube-prom-coredns",
    "kube-system/cerus-prometheus-kube-prom-kube-controller-manager",
    "kube-system/cerus-prometheus-kube-prom-kube-proxy",
    "kube-system/cerus-prometheus-kube-prom-kube-scheduler"
  ],
  "ServiceAccount/v1": [
    "cerusbots/cerus-prometheus-kube-prom-alertmanager",
    "cerusbots/cerus-prometheus-kube-prom-blackbox-exporter",
    "cerusbots/cerus-prometheus-kube-prom-operator",
    "cerusbots/cerus-prometheus-kube-prom-prometheus",
    "cerusbots/cerus-prometheus-kube-state-metrics",
    "cerusbots/cerus-prometheus-node-exporter"
  ],
  "ServiceMonitor.monitoring.coreos.com/monitoring.coreos.com/v1": [
    "cerusbots/cerus-prometheus-kube-prom-alertmanager",
    "cerusbots/cerus-prometheus-kube-prom-apiserver",
    "cerusbots/cerus-prometheus-kube-prom-coredns",
    "cerusbots/cerus-prometheus-kube-prom-kube-controller-manager",
    "cerusbots/cerus-prometheus-kube-prom-kube-proxy",
    "cerusbots/cerus-prometheus-kube-prom-kube-scheduler",
    "cerusbots/cerus-prometheus-kube-prom-kubelet",
    "cerusbots/cerus-prometheus-kube-prom-operator",
    "cerusbots/cerus-prometheus-kube-prom-prometheus",
    "cerusbots/cerus-prometheus-kube-state-metrics",
    "cerusbots/cerus-prometheus-node-exporter"
  ]
}
```

**Comment by RossComputerGuy** [comment]
Alright, I've signed the commits and I've also fixed any conflicts

**Review by nicks — APPROVED** [review]
thanks! LGTM

**Pull request merged**